app/services/delivery_request_service.rb

```diff
@@ -5,31 +5,34 @@ class DeliveryRequestService
     "delay" => DelayProvider,
   }.freeze
 
-  attr_reader :provider_name, :provider, :subject_prefix, :overrider
-
-  def initialize(config: EmailAlertAPI.config.email_service)
+  def initialize(email:, config: EmailAlertAPI.config.email_service)
+    @email = email
+    @attempt_id = SecureRandom.uuid
     @provider_name = config.fetch(:provider).downcase
-    @provider = PROVIDERS.fetch(provider_name)
     @subject_prefix = config.fetch(:email_subject_prefix)
     @overrider = EmailAddressOverrider.new(config)
   end
 
   def self.call(*args)
-    new.call(*args)
+    new(*args).call
```

Comment:
Out-of-scope: looking at the other services, we're quite close to being able to give them all a common superclass. Do you think an `ApplicationService` makes sense for this repo?

Reply:
Yeah, if it's consistent enough to do it that'd be a good idea to at least try prevent them going experimental again later.